src/sagemaker_server.cc

```diff
@@ -670,6 +732,35 @@ SagemakerAPIServer::SageMakerMMEUnloadModel(
 
   HandleRepositoryControl(req, "", model_name, "unload");
 
+  /*Note: Model status check is repo-specific and therefore must be run before
+   * unregistering the repo, else the model information is lost*/
+  bool is_model_unavailable = false;
+  uint32_t unload_time_in_secs = 0;
+
+  LOG_VERBOSE(1) << "Using Model Repository Index during UNLOAD to check for "
+                    "status of model: "
+                 << model_name;
+
+  /* Wait for the model to be completely unloaded. SageMaker waits a maximum of
+  360 seconds for the UNLOAD request to timeout. Setting a limit of 350 seconds
+  for Triton unload.*/
+  while (!is_model_unavailable && unload_time_in_secs < UNLOAD_TIMEOUT_SECS_) {
+    is_model_unavailable = SageMakerMMEUnloadModelCheckStatus(model_name);
+    sleep(1);
```

Comment:
Is it ok to block the thread with `sleep` like this? Or do we need to do some kind of shenanigans with event loop to check again after a timeout without blocking the thread?